SpatialGDK/Source/SpatialGDK/Private/Tests/RPCServiceTest.cpp

```diff
@@ -0,0 +1,549 @@
+		SpatialGDK::EntityComponentId(RPCTestEntityId_1, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID),
+		ClientComponentData,
+		ClientAuth);
+
+	AddEntityComponentToStaticComponentView(*StaticComponentView,
+		SpatialGDK::EntityComponentId(RPCTestEntityId_1, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID),
+		ServerAuth);
+
+	bool bAttemptedRPCExtraction = false;
+	bool bPayloadsMatch = true;
+	ExtractRPCDelegate RPCDelegate = ExtractRPCDelegate::CreateLambda([&bAttemptedRPCExtraction, &bPayloadsMatch](Worker_EntityId EntityId, ERPCType RPCType, const SpatialGDK::RPCPayload& Payload) {
+		bAttemptedRPCExtraction = true;
+		bPayloadsMatch &= CompareRPCPayload(Payload, SimplePayload);
+		bPayloadsMatch &= EntityId == RPCTestEntityId_1;
+		return true;
+	});
+
+	SpatialGDK::SpatialRPCService RPCService = CreateRPCService({ RPCTestEntityId_1 }, SERVER_AUTH, RPCDelegate, StaticComponentView);
+	RPCService.ExtractRPCsForEntity(RPCTestEntityId_1, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID);
+
+	TestTrue("Extracted RPCs match expected payloads", (bAttemptedRPCExtraction && bPayloadsMatch));
+	return true;
+}
+
+RPC_SERVICE_TEST(GIVEN_receiving_an_rpc_WHEN_return_false_from_extract_callback_THEN_some_rpcs_persist_on_component)
+{
+	USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+
+	Worker_Authority ClientAuth = GetClientAuthorityFromRPCEndpointType(SERVER_AUTH);
+	Worker_Authority ServerAuth = GetServerAuthorityFromRPCEndpointType(SERVER_AUTH);
+
+	Schema_ComponentData* ClientComponentData = Schema_CreateComponentData();
+	Schema_Object* ClientSchemaObject = Schema_GetComponentDataFields(ClientComponentData);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 1, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 2, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 3, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 4, SimplePayload);
+	SpatialGDK::RPCRingBufferUtils::WriteRPCToSchema(ClientSchemaObject, ERPCType::ClientReliable, 5, SimplePayload);
+
+	AddEntityComponentToStaticComponentView(*StaticComponentView,
+		SpatialGDK::EntityComponentId(RPCTestEntityId_1, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID),
+		ClientComponentData,
+		ClientAuth);
+
+	AddEntityComponentToStaticComponentView(*StaticComponentView,
+		SpatialGDK::EntityComponentId(RPCTestEntityId_1, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID),
+		ServerAuth);
+
+	const static int8 MaxRPCsToProccess = 2;
+	int8 RPCsToProccess = MaxRPCsToProccess;
+	ExtractRPCDelegate RPCDelegate = ExtractRPCDelegate::CreateLambda([&RPCsToProccess](Worker_EntityId EntityId, ERPCType RPCType, const SpatialGDK::RPCPayload& Payload) {
+		bool bKeepProcessing = (RPCsToProccess != 0);
```

Comment:
I think this would be neater as
```suggestion
		--RPCsToProcess;
		return RPCsToProcess >= 0;
```

Reply:
Done